extensions/wikia/AchievementsII/Ach_setup.php

```diff
@@ -158,25 +155,25 @@ function Ach_ArticleSaveComplete(	&$article, &$user, $text,
 	return true;
 }
 
-function Ach_GetHTMLAfterBody($skin, &$html) {
+function Ach_GetHTMLAfterBody( Skin $skin, &$html ): bool {
 	wfProfileIn(__METHOD__);
 
-	global $wgOut, $wgUser;
+	$user = $skin->getUser();
 
-	if($wgUser->isLoggedIn() && !($wgUser->getGlobalPreference( 'hidepersonalachievements' ))) {
+	if ($user->isLoggedIn() && !( $user->getGlobalPreference( 'hidepersonalachievements' ) ) ) {
 		if( $skin->getTitle()->isSpecial( 'MyHome' ) ) {
 			$awardingService = new AchAwardingService();
-			$awardingService->awardCustomNotInTrackBadge($wgUser, BADGE_WELCOME);
+			$awardingService->awardCustomNotInTrackBadge( $user, BADGE_WELCOME );
 		}
 
-		if((!empty($_SESSION['achievementsNewBadges']) || 5 == rand(1, 20)) && get_class(RequestContext::getMain()->getSkin()) != 'SkinMonobook') {
+		if ( ( !empty( $_SESSION['achievementsNewBadges'] ) || 5 == rand( 1, 20 ) ) && $skin->getSkinName() !== 'monobook' ) {
```

Comment:
Hmm, the closest I could think of would be to introduce some `Skin::is( string $skinName ): boolean;` method, which would essentially call this under the hood, but I'm not sure if it'd be worth it.